cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/MetadataService.java

```diff
@@ -60,7 +60,7 @@
     NettyHttpService.Builder builder = new CommonNettyHttpServiceBuilder(cConf, Constants.Service.METADATA_SERVICE)
       .setHttpHandlers(handlers)
       .setExceptionHandler(new HttpExceptionHandler())
-      .setHandlerHooks(Collections.singleton(new MetricsReporterHook(metricsCollectionService,
+      .setHandlerHooks(Collections.singleton(new MetricsReporterHook(cConf, metricsCollectionService,
```

Comment:
Is it possible to pass cConf using Guice, instead of changing the constructor everywhere?

Reply:
The creation of the `MetricsReporterHook` is not through guice, mainly because of the different service name for different service. Technically we can create a hook factory with Assisted inject, but it seems to be an overkill in this particular case. However, I think we can move this `setHandlerHooks` call into the `CommonNettyHttpServiceBuilder` so that it is also added for all services.